File: 0713-Subarray-Product-Less-Than-K.py
class Solution(object):
    def numSubarrayProductLessThanK(self, nums, k):
        if k == 0:
            return 0
        
        n = len(nums)
        subArrayCount = 0
        j = 0
        currentProduct = 1
        
        for i in range(n):
            currentProduct *= nums[i]
            while j <= i and currentProduct >= k:
                currentProduct //= nums[j]
                j += 1
            subArrayCount += (i - j + 1)
            
        return subArrayCount

File: 0713-Subarray-Product-Less-Than-K.js
var numSubarrayProductLessThanK = function(nums, k) {
    if (k === 0) {
        return 0;
    }
    
    let n = nums.length;
    let subArrayCount = 0;
    let j = 0;
    let currentProduct = 1;
    
    for (let i = 0; i < n; i++) {
        currentProduct *= nums[i];
        while (j <= i && currentProduct >= k) {
            currentProduct /= nums[j];
            j ++;
        }
        subArrayCount += (i - j + 1);
    }
    
    return subArrayCount;
};